### ISSUE TITLE
StandardScaler needs an inverse transform function

### ISSUE THREAD

**Issue by callmekatootie** [opensource9ja/danfojs#222]
**Is your feature request related to a problem? Please describe.**

I normalized the data using [StandardScaler](https://danfo.jsdata.org/api-reference/general-functions/danfo.standardscaler). I then trained a tensorflowjs model. I now wish to predict. In order to carry this out, I need to first normalize and then use tensorflowjs's model predict() function - and then get the un-normalized prediction.

I assume to normalize that data that I will use for prediction, I re-use the same scaler to transform() (without running fit() first since we have already used fit() before during training - please let me know if I am wrong). But to get the un-normalized prediction, I need a reverse transform function

**Describe the solution you'd like**

sklearn appears to have an [inverse_transform()](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html?highlight=standardscaler#sklearn.preprocessing.StandardScaler.inverse_transform) function that needs to exist in Danfojs too

**Describe alternatives you've considered**

I am still trying to figure out how to get the inverse transform implemented, since I do not have the mean or standard deviation information with me. Going through Danfojs's source code to figure if it can be implemented or if I need to set up my own scaler and reverse transform functions

**Comment by callmekatootie** [comment]
Related https://github.com/opensource9ja/danfojs/issues/103

**Comment by callmekatootie** [comment]
Related - On the same note, would like to also see the constructor for StandardScaler (or MinMaxScaler) to accept a mean and standard deviation values, that would have been earlier calculated - so that it can be used to transform data again using the same mean and std

**Comment by steveoni** [comment]
@callmekatootie sorry for the late reply. But if you will like to implement this, I can point out somethings to you:

For `StandardScalar` we have `this.mean` and `this.std`, with this variable we can reconstruct the normalized data back.

This is how the normalized data is generated

```js
output_data = tensor_data.sub(this.mean).div(this.std).arraySync();
```

to get the unnormalized data back we can do it like this:

```js
normalized = tf.tensor(output_data)
unnormalized = output_data.mul(this.std).add(this.mean).arraySync()
```

The same thing applies to MinMaxScalar.

**Comment by callmekatootie** [comment]
Yep. That's how I did it too. It would be nice-to-have this as a method though...

**Comment by steveoni** [comment]
@callmekatootie that's true. you can help add your implementation as a method.

**Comment by risenW** [comment]
@steveoni will yu work on this?

**Comment by callmekatootie** [comment]
@risenW @steveoni I can work on it. I have a question though - Should I run the build after I carry out the changes, or is that happening automatically, say through a github action or something else? I see that the dist folder is part of the commit so unsure if I need to do that prior to submitting the PR or it happens automatically post merge...

**Comment by callmekatootie** [comment]
Also,

- which nodejs version should I verify my changes with, for both the browser and node versions of the package
- shouldn't I commit the `package-lock.json` - I don't see it right now but I _think_ we should?

**Comment by risenW** [comment]
> @risenW @steveoni I can work on it. I have a question though - Should I run the build after I carry out the changes, or is that happening automatically, say through a github action or something else? I see that the dist folder is part of the commit so unsure if I need to do that prior to submitting the PR or it happens automatically post merge...

Awesome! 
>  Should I run the build after I carry out the changes, or is that happening automatically, say through a github action or something else?

Yes, you can run a build to test locally, pushing the build/dist folder to Github is not required. we have configured a bunch of script in package.json for you to do that already. Also, you can add some tests in the corresponding test folder here. 

Also, see the contributing guide [here](https://danfo.jsdata.org/contributing-guide), it will help you get started. If you need any information, drop a comment here, and we'll gladly help out.

**Comment by risenW** [comment]
> which nodejs version should I verify my changes with, for both the browser and node versions of the package

Our [contribution guide](https://danfo.jsdata.org/contributing-guide) explains this better. We have two separate folders for the Node and Browser version.

**Comment by callmekatootie** [comment]
PR ready for review - https://github.com/opensource9ja/danfojs/pull/271
